lib/Serverless.test.js

```diff
@@ -358,4 +359,49 @@ describe('Serverless [new tests]', () => {
         expect(serverless.isLocalStub).to.not.exist;
       }));
   });
+
+  describe('When .env file is available', () => {
+    let servicePath;
+
+    before(async () => {
+      servicePath = (
+        await fixtures.setup('function', {
+          configExt: {
+            useDotenv: true,
+            custom: {
+              fromDefaultEnv: '${env:DEFAULT_ENV_VARIABLE}',
+              fromStageEnv: '${env:STAGE_ENV_VARIABLE}',
+            },
+          },
+        })
+      ).servicePath;
+
+      const defaultFileContent = 'DEFAULT_ENV_VARIABLE=valuefromdefault';
+      writeFileSync(path.join(servicePath, '.env'), defaultFileContent);
```

Comment:
Let's use _async_ function (and rely on `fs.promises` directly). 

(Unfortunately in Framework we have some not great old patterns that dates Node.js v4 and before, we will slowly recover from that, and ideally if new code is free from them)